packages/@aws-cdk/aws-ec2/lib/cfn-init.ts

```diff
@@ -125,14 +125,24 @@ export class CloudFormationInit {
     // To identify the resources that have the metadata and where the signal
     // needs to be sent, we need { region, stackName, logicalId }
     let resourceLocator = `--region ${Aws.REGION} --stack ${Aws.STACK_NAME} --resource ${attachedResource.logicalId}`;
+    let notifyResourceLocator = `--region ${Aws.REGION} --stack ${
+      Aws.STACK_NAME
+    } --resource ${
+      attachOptions.signalResource
+        ? attachOptions.signalResource.logicalId
+        : attachedResource.logicalId
+    }`;
```

Comment:
The formatting here is a bit odd. I think this will read a bit cleaner.

```suggestion
    const signalResource = attachOptions.signalResource?.logicalId ?? attachedResource.logicalId;
    let notifyResourceLocator = `--region ${Aws.REGION} --stack ${Aws.STACK_NAME} --resource ${signalResource}`;
```